state_machine/drivers/pycubedmini/lib/pycubed.py

```diff
@@ -54,47 +57,86 @@ def __init__(self):
         self._stat = {}
         self.BOOTTIME = const(self.timeon)
         self.hardware = {
-            'IMU':    False,
-            'Radio':  False,
+            'I2C1': False,
+            'I2C2': False,
+            'I2C3': False,
+            'SPI': False,
             'SDcard': False,
-            'GPS':    False,
-            'WDT':    False,
-            'Sun':    False,
-            'Coils':  False,
-            'BurnWire': False,
+            'Neopixel': False,
+            'IMU': False,
+            'Radio': False,
+            'Sun -Y': False,
+            'Sun -Z': False,
+            'Sun -X': False,
+            'Sun +Y': False,
+            'Sun +Z': False,
+            'Sun +X': False,
+            'Coil X': False,
+            'Coil Y': False,
+            'Coil Z': False,
+            'Burn Wire 1': False,
+            'Burn Wire 2': False,
+            'WDT': False  # Watch Dog Timer pending
         }
         self.micro = microcontroller
-
         self.data_cache = {}
         self.filenumbers = {}
         self.vlowbatt = 3.5
         self.debug = True
+        self._vbatt = analogio.AnalogIn(board.BATTERY)  # Define battery voltage
+
+        # Define and initialize hardware
+        self.__init_i2c__()
```

Comment:
This is a style thing but I think the convention is that functions starting and ending with two underscores `__function__` are python built in functions that have "special" meaning, such as being the object constructor. Since these are not built in functions (unlike `__init__`), these should all start with a single underscore and end with no underscores (`_init_i2c()`). 

See: https://dbader.org/blog/meaning-of-underscores-in-python

Reply:
I like this idea